pkg/parser/ast/misc.go

```diff
@@ -1493,6 +1493,9 @@ type UserSpec struct {
 	User    *auth.UserIdentity
 	AuthOpt *AuthOption
 	IsRole  bool
+
+	// DefaultAuthPlugin is used to store the values of default_authentication_plugin
+	DefaultAuthPlugin string
```

Comment:
How about to remove `DefaultAuthPlugin` in `UserSpec` and move `EncodedPassword` to executor? Seems `EncodedPassword` is only used by executor and it's better to keep `UserSpec` to serve for parser only.

Reply:
Update in https://github.com/pingcap/tidb/pull/57391/commits/50a049aea2aa808370a184d7afb0d11cb88dbffa